src/DevChatter.Bot.Core/Automation/AutomatedActionFactory.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using DevChatter.Bot.Core.Games.DealNoDeal;
+using DevChatter.Bot.Core.Systems.Chat;
+using DevChatter.Bot.Core.Util;
+
+namespace DevChatter.Bot.Core.Automation
+{
+    public class AutomatedActionFactory : IAutomatedActionFactory
+    {
+        private readonly DealNoDealGame _dealNoDealGame;
+        private readonly IChatClient _chatClient;
+
+        public AutomatedActionFactory(DealNoDealGame dealNoDealGame, IChatClient chatClient)
+        {
+            _dealNoDealGame = dealNoDealGame;
+            _chatClient = chatClient;
+        }
+        public IIntervalAction GetIntervalAction(DealNoDealGameState gameState)
+        {
+            if (gameState == DealNoDealGameState.CHOSING_STARTING_BOXES)
```

Comment:
Doesn't require a change, but I'd recommend using a `switch` statement here, since this is an enum. It's possible that the C# compiler is smart enough to optimize this to a switch for you, but I'm not sure if it does for this code.

Reply:
Change made. aeab4d34e1bfd2de21d826ffc555f2ce2b995870 (I hope this is how you link commits 😄 )

Reply:
It works!